rmqsrc/encode.nim

```diff
@@ -20,21 +23,28 @@ proc encode(v: string): seq[char] =
 proc encode(vnode: ValueNode): seq[char] =
   result = newSeq[char]()
   case vnode.valueType
+  of vtBool:
+    result.add(encode(vnode.boolValue.uint8))
```

Comment:
I don't think you need to declare `result` and then add. I think you can say `result = @[foo]` in every case.